service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/dataset/CompleteTableCopyJobsStep.java

```diff
@@ -0,0 +1,100 @@
+package bio.terra.workspace.service.resource.controlled.flight.clone.dataset;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.service.crl.CrlService;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.job.JobMapKeys;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.google.api.services.bigquery.Bigquery;
+import com.google.api.services.bigquery.model.Job;
+import com.google.api.services.bigquery.model.JobReference;
+import com.google.api.services.bigquery.model.TableReference;
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CompleteTableCopyJobsStep implements Step {
+  private static final Logger logger = LoggerFactory.getLogger(CompleteTableCopyJobsStep.class);
+  private final CrlService crlService;
+
+  public CompleteTableCopyJobsStep(CrlService crlService) {
+    this.crlService = crlService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+    final FlightMap workingMap = flightContext.getWorkingMap();
+    final Map<String, String> tableToJobId =
+        workingMap.get(ControlledResourceKeys.TABLE_TO_JOB_ID_MAP, new TypeReference<>() {});
+    final AuthenticatedUserRequest userRequest =
+        flightContext
+            .getInputParameters()
+            .get(JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
+    final Bigquery bigQueryClient = crlService.createNakedBigQueryClient(userRequest);
+    try {
+      for (Map.Entry<String, String> entry : tableToJobId.entrySet()) {
+        final TableReference tableReference = tableFqIdToReference(entry.getKey());
+        final JobReference jobReference = jobFqIdToReference(entry.getValue());
+
+        // wait for job to complete
+        while (true) {
+          final Job job =
+              bigQueryClient
+                  .jobs()
+                  .get(jobReference.getProjectId(), jobReference.getJobId())
+                  .setLocation(
+                      jobReference.getLocation()) // returns NOT_FOUND unless location is specified
+                  .execute();
+          final String jobState = job.getStatus().getState();
+          logger.debug("Table {} is {}", tableReference.getTableId(), jobState);
+          if ("DONE".equals(jobState)) {
+            break;
+          }
+          TimeUnit.SECONDS.sleep(5);
```

Comment:
Maybe a static constant for the sleep time?
What led you to pick 5 seconds?

Reply:
It was the lowest I've seen used. We could go smaller. Most in-region copies will happen almost instantly I'm hoping.

Reply:
Do we have a reusable exponential backoff loop construct?

Reply:
implemented